include/nbl/asset/IRenderpass.h

```diff
@@ -13,102 +13,556 @@
+                        };
+                        struct ResolveMode
+                        {
+                            E_RESOLVE_MODE depth = NONE;
+                            E_RESOLVE_MODE stencil = NONE;
+                        } resolveMode;
+                    };
+                    using SColorAttachmentRef = SRenderAttachmentRef<IImage::E_LAYOUT>;
+
+
+                    auto operator<=>(const SSubpassDescription&) const = default;
+
+                    inline bool valid() const
+                    {
+                        if (!depthStencilAttachment.valid())
+                            return false;
+                        for (auto i=0u; i<MaxColorAttachments; i++)
+                        {
+                            if (!colorAttachments[i].valid())
+                                return false;
+                            // https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkSubpassDescription2.html#VUID-VkSubpassDescription2-pDepthStencilAttachment-04440
+                            if (depthStencilAttachment.render.used() && depthStencilAttachment.render.attachmentIndex==colorAttachments[i].render.attachmentIndex)
+                                return false;
+                        }
+                        bool invalid = false;
+                        auto attachmentValidVisitor = [&invalid](const auto& ref)->bool
+                        {
+                            if (!ref.valid())
+                            {
+                                invalid = true;
+                                return false;
+                            }
+                            return true;
+                        };
+                        visitTokenTerminatedArray(inputAttachments,InputAttachmentsEnd,attachmentValidVisitor);
+                        if (invalid)
+                            return false;
+                        // https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkSubpassDescription2.html#VUID-VkSubpassDescription2-flags-03076
+                        if (flags.hasFlags(E_SUBPASS_DESCRIPTION_FLAGS::PER_VIEW_POSITION_X_ONLY_BIT) && !flags.hasFlags(E_SUBPASS_DESCRIPTION_FLAGS::PER_VIEW_ATTRIBUTES_BIT))
+                            return false;
+                        return true;
+                    }
+
+
+                    //! Field ordering prioritizes ergonomics
+                    static inline constexpr auto MaxColorAttachments = 8u;
+                    SColorAttachmentRef colorAttachments[MaxColorAttachments] = {};
+
+                    SDepthStencilAttachmentRef depthStencilAttachment = {};
+
+                    constexpr static inline SInputAttachmentRef InputAttachmentsEnd = {};
+                    const SInputAttachmentRef* inputAttachments = &InputAttachmentsEnd;
+
+                    // The arrays pointed to by this array must be terminated by `AttachmentUnused` value, which implicitly satisfies:
```

Comment:
I think a similar statement should be commented for the attachments above?
seems like we removed the `count` var for all of them and now have to use `IRenderpass::SCreationParams::SSubpassDescription::SXXXAttachmentRefEnd` at the end of the array for termination

Reply:
yep, ok put matching comments everywhere